Bug: Cannot use aliased imports with 'using' statement

## Description

When importing a module with an alias using the syntax `import alias@module`, the `using` statement fails to recognize the alias and reports that the module is not imported, even though it was imported on a previous line.

## Example EZ Code

```ez
import @std
import arr@arrays

using std
using arr  // Error: "cannot use module 'arr' before importing it"

do main() {
  some_thing()
}

const People struct {
  names [string]
  ages [int]
}

do some_thing() -> People {
  temp people = new(People)
  temp names [string] = {"Bob", "Bill", "Angela"}

  for_each n in names {
    push(people.names, n)
  }

  println(people.names)
}
```

## Expected Behavior

The `using arr` statement should successfully bring the aliased `arrays` module into scope, allowing access to functions like `push()`, `len()`, etc. without the `arr.` prefix.

The program should:
1. Import `@arrays` with alias `arr` ✓
2. Recognize `arr` as a valid imported module for the `using` statement ✓
3. Execute successfully with array functions available ✓

## Actual Behavior

Parser/evaluator fails with error:

```
error[E1003]: cannot use module 'arr' before importing it
  --> ./main.ez:5:1
   |
5 | using arr
   | ^^^^^ invalid syntax
   |

summary: generated 1 error
```

The error is misleading because:
- `arr` **was** imported on line 2: `import arr@arrays`
- The error says "before importing" but import happens before the using statement
- The actual issue is that the alias isn't being tracked/recognized

## Root Cause

The import system likely has a mismatch between how it:
1. **Stores** the import: Probably using the module name `arrays` as the key
2. **Looks up** for using: Trying to find `arr` but can't locate it

Looking at `pkg/interpreter/evaluator.go` around lines 130-146:

```go
case *ast.ImportStatement:
    // Validate that the module exists
    if !isValidModule(node.Module) {
        return newErrorWithLocation("E5002", node.Token.Line, node.Token.Column,
            "module '%s' not found", node.Module)
    }

    alias := node.Alias
    if alias == "" {
        alias = node.Module
    }
    env.Import(alias, node.Module)
    return NIL
```

And the using statement validation (lines 148-159):

```go
case *ast.UsingStatement:
    // Bring the module(s) functions into scope (function-scoped using)
    for _, module := range node.Modules {
        alias := module.Value
        // Verify the module was imported
        if _, ok := env.GetImport(alias); !ok {
            return newErrorWithLocation("E3005", node.Token.Line, node.Token.Column,
                "cannot use '%s': module not imported", alias)
        }
        env.Use(alias)
    }
    return NIL
```

The issue is likely that when parsing `import arr@arrays`, the parser might be setting the alias incorrectly, or the environment's import tracking isn't properly storing the alias→module mapping.

## Implementation Notes

### Check Parser (`pkg/parser/parser.go`)

Verify that when parsing `import arr@arrays`:
- `ImportStatement.Module` = "arrays"
- `ImportStatement.Alias` = "arr"

The parser might be reversing these or not setting them correctly.

### Check Environment (`pkg/interpreter/environment.go`)

The environment's `Import()` method should:
- Store the mapping: `alias → module`
- When `GetImport(alias)` is called, return the module name

Example expected behavior:
```go
env.Import("arr", "arrays")  // Store: arr → arrays
module, ok := env.GetImport("arr")  // Should return: "arrays", true
```

### Check Using Statement Validation

The using statement validation should look up the alias in imports. If `env.GetImport("arr")` returns `("arrays", true)`, then `using arr` should succeed.

## Workaround

Currently, users must use the non-aliased syntax:
```ez
import @arrays

using arrays
```

Or call functions with the full prefix (if that works):
```ez
import arr@arrays

do main() {
    arr.push(list, item)  // Use explicit prefix instead of 'using'
}
```

## Test Cases to Add

```ez
// Test 1: Basic alias with using
import arr@arrays
using arr
// Should work

// Test 2: Multiple aliases
import arr@arrays
import str@strings
using arr
using str
// Both should work

// Test 3: Alias and non-alias mixed
import @std
import arr@arrays
using std
using arr
// Both should work

// Test 4: Function-scoped using with alias
import arr@arrays

do main() {
    using arr
    push(list, 1)  // Should work in function scope
}
```

## Notes

- This affects both file-scoped and function-scoped `using` statements
- Error message should be clearer: "alias 'arr' not found in imports"
- Related to the import system and environment tracking
- May also affect member access like `arr.push()` (needs verification)
- Will be especially important once feature #78 (`import & use arr@arrays`) is implemented

--- a/pkg/parser/parser.go
+++ b/pkg/parser/parser.go
@@ -391,14 +391,12 @@ func (p *Parser) ParseProgram() *Program {
 		if stmt != nil {
 			program.Statements = append(program.Statements, stmt)
 
-			// Track imported modules
+			// Track imported modules by both alias and module name
 			if importStmt, ok := stmt.(*ImportStatement); ok {
-				// Extract module name (handle both @std and alias@std)
-				moduleName := importStmt.Module
-				if strings.HasPrefix(moduleName, "@") {
-					moduleName = moduleName[1:] // Remove @ prefix
-				}
-				importedModules[moduleName] = true
+				// Track the alias so "using arr" works with "import arr@arrays"
+				importedModules[importStmt.Alias] = true
+				// Also track the module name so "using arrays" still works
+				importedModules[importStmt.Module] = true
 			}
 		}
 		p.nextToken()

